pkg/model/postgresql_service.go

```diff
@@ -12,16 +12,21 @@ import (
 
 func getSpec(dbSecret *v1.Secret, serviceTypeExternal bool) v1.ServiceSpec {
 	spec := v1.ServiceSpec{}
+	isIPAddress := dbSecret != nil && dbSecret.Data[DatabaseSecretExternalAddressProperty] != nil && IsIP(dbSecret.Data[DatabaseSecretExternalAddressProperty])
 
-	if serviceTypeExternal {
+	if serviceTypeExternal && !isIPAddress {
 		spec.Type = v1.ServiceTypeExternalName
 		spec.Selector = nil
 		spec.ExternalName = GetExternalDatabaseHost(dbSecret)
 	} else {
 		spec.Type = v1.ServiceTypeClusterIP
-		spec.Selector = map[string]string{
-			"app":       ApplicationName,
-			"component": PostgresqlDeploymentComponent,
+		if !isIPAddress {
```

Comment:
Please just leave a short comment here explaining why are we not setting selectors for IP addresses (you can mention the JIRA ID).

Reply:
:+1: